src/normal.c

```diff
@@ -2562,7 +2562,14 @@ nv_z_get_count(cmdarg_T *cap, int *nchar_arg)
 	if (nchar == K_DEL || nchar == K_KDEL)
 	    n /= 10;
 	else if (VIM_ISDIGIT(nchar))
+	{
+	    if (n > LONG_MAX/10)
+	    {
+		clearopbeep(cap->oap);
+		break;
+	    }
 	    n = n * 10 + (nchar - '0');
+	}
```

Comment:
It's probably been over 30 years since I was checking for "C" code overflow but this looks suspicious. I don't know what the above code is doing or what values might actually be encountered, but the precondition check 
```
n > LONG_MAX/10
``` 
looks wrong. Notice that there is overflow after the first check in the following example.

```
#include <stdio.h>
#include <limits.h>

#define SAFE_SPACE 100

int main(int ac, char *av[]) {
    int nchar = '9' - '0';  // Not sure what legit values nchar can take.

    long n = LONG_MAX / 10;
    if (n > LONG_MAX / 10)
        printf("BREAK\n");
    else
        printf("%ld\n", n * 10 + ('9' - '0'));

    n = LONG_MAX / 10;
    if (n > LONG_MAX / 10 - SAFE_SPACE)
        printf("BREAK\n");
    else
        printf("%ld\n", n * 10 + ('9' - '0'));
}
```
```
-9223372036854775807
BREAK
```

Reply:
true. But I think this will break already in the previous iteration of the loop

Reply:
okay, should be fixed now